File: mqtt_send.py
import amqp_rpc as rpc
import pika
import time

MQ_USER = 'onionCore'
MQ_PASS = 'p'
MQ_HOST = 'mqtt.onion.io'

credentials = pika.PlainCredentials(MQ_USER, MQ_PASS)
parameters = pika.ConnectionParameters(credentials=credentials, host=MQ_HOST, virtual_host='/')

sendConnection = pika.BlockingConnection(parameters)
sendChannel = sendConnection.channel()

@rpc.register
def IF_MQTT_SEND(params):
    deviceId = params['deviceId']
    cmd = str(params['cmd'])
    sendChannel.basic_publish(exchange='amq.topic', routing_key='.%s'%deviceId, body=cmd)

rpc.loop()

File: mqtt_send.js
var rpc = require('./amqp-rpc/amqp_rpc');
var amqp = require('./amqp-rpc/node_modules/amqplib');

var log = function(msg){
    process.stdout.write("test::amqp_rpc.js: ");
    console.log (msg);
}

var mqServerUrl = 'amqp://onionCore:p@mqtt.onion.io';

rpc.register('IF_MQTT_SEND', function(params, callback){
    var key = '.'+params.deviceId
    var cmd = params.cmd.toString()
    log(key);

    var open = amqp.connect(mqServerUrl);
    open.then(function(conn) {
        conn.createChannel().then(function(ch) {
            ch.publish('amq.topic', key, new Buffer(cmd));
            conn.close();
        });
    });

    log(params);

    callback({});
});


